File: Api/Models/Request.cs
namespace Api.Models
{
    public class Request
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int TeamId { get; set; }
        public int StatusId { get; set; } = 1;
        public DateTime RequestDate { get; set; } = DateTime.Now;

        public User? User { get; set; }
        public Team? Team { get; set; }
        public Status? Status { get; set; }

        public void Decline()
        {
            StatusId = 3;
        }
        public void Accept()
        {
            StatusId = 2;
        }
    }
}

File: Client/src/app/models/request.ts
import { Status } from "./status";
import { Team } from "./team";
import { User } from "./user";

export class Request {
    id: number = 0;
    userId: number = 0;
    teamId: number = 0;
    statusId: number = 0;
    requestDate?: Date;
    user?: User;
    team?: Team;
    status?: Status;
}